lnrpc/walletrpc/walletkit_server.go

```diff
@@ -1215,3 +1230,187 @@ func (w *WalletKit) FinalizePsbt(_ context.Context,
 		RawFinalTx: finalTxBytes.Bytes(),
 	}, nil
 }
+
+// marshalWalletAccount converts the properties of an account into its RPC
+// representation.
+func marshalWalletAccount(account *waddrmgr.AccountProperties) (*Account, error) {
+	var addrType AddressType
+	switch account.KeyScope {
+	case waddrmgr.KeyScopeBIP0049Plus:
+		if account.AddrSchema == nil {
+			addrType = AddressType_NESTED_WITNESS_PUBKEY_HASH
+		} else {
+			addrType = AddressType_HYBRID_NESTED_WITNESS_PUBKEY_HASH
```

Comment:
? 

Could use a comment explaining it perhaps?

Reply:
Shouldn't this read the `AddrSchema` instead and map it to one of the enums?

Reply:
Fixed. Noticed the address types were assigned incorrectly as well.